auth/service.cc

```diff
@@ -437,23 +438,23 @@ future<> create_role(
         std::string_view name,
         const role_config& config,
         const authentication_options& options) {
-    return ser.underlying_role_manager().create(name, config).then([&ser, name, &options] {
-        if (!auth::any_authentication_options(options)) {
-            return make_ready_future<>();
-        }
-
-        return futurize_invoke(
-                &validate_authentication_options_are_supported,
-                options,
-                ser.underlying_authenticator().supported_options()).then([&ser, name, &options] {
-            return ser.underlying_authenticator().create(name, options);
-        }).handle_exception([&ser, name](std::exception_ptr ep) {
-            // Roll-back.
-            return ser.underlying_role_manager().drop(name).then([ep = std::move(ep)] {
-                std::rethrow_exception(ep);
-            });
-        });
-    });
+    co_await ser.underlying_role_manager().create(name, config);
+    if (!auth::any_authentication_options(options)) {
+        co_return;
+    }
+    std::exception_ptr ep;
+    try {
+        validate_authentication_options_are_supported(options,
+                ser.underlying_authenticator().supported_options());
+        co_await ser.underlying_authenticator().create(name, options);
+    } catch (...) {
```

Comment:
We're really interested in catching exceptions from validate(), not create(). But it's a faithful conversion.

Reply:
in this case from create too because of the rollback below, in auth-v2 yes as rollback will be removed

Reply:
If create() throws, the role will not have been created, so the drop is unnecessary. But I don't suggest to change anything.

Reply:
yes, but there are 2 create() in this func,  catch is for the second one, and rollback for the first one (yes, confusing, sad that auth-v1 needs to stay)

Reply:
You're right of course. In two years we can drop v1.